src/config.js

```diff
@@ -114,6 +114,26 @@ export function newConfig() {
         return this._customPriceBucket;
       },
 
+      _mediaTypePriceGranularity: {},
+      get mediaTypePriceGranularity() {
+        return this._mediaTypePriceGranularity;
+      },
+      set mediaTypePriceGranularity(val) {
+        this._mediaTypePriceGranularity = Object.keys(val).reduce((aggregate, item) => {
+          if (validatePriceGranularity(val[item])) {
+            if (typeof val === 'string') {
```

Comment:
looks like this code was copied from the function above around line ~90. Can we combine it/make more generic?